File: runtime/go/units/wasi/fd_filestat_set_times.go
// requires: wasi/resolve_times, wasi/errno_fs
// Set the access/modification times of a file fd. fstflags is
// validated first (so a bad combination is EINVAL even though nothing else
// runs); os.Chtimes leaves a field whose zero time.Time was returned untouched.
func (w *WASI) wasi_fd_filestat_set_times(fd uint32, atim, mtim uint64, fstflags uint32) uint32 {
    f, ok := w.fds[fd].(*os.File)
    if !ok {
        return wasiBadf
    }
    atime, mtime, e := w.resolve_times(atim, mtim, fstflags)
    if e != wasiOk {
        return e
    }
    if err := os.Chtimes(f.Name(), atime, mtime); err != nil {
        return w.fs_errno(err)
    }
    return wasiOk
}

File: crates/dewasm-backend-python/units/wasi/fd_filestat_set_times.py
# requires: wasi/fst_times
def wasi_fd_filestat_set_times(self, fd, atim, mtim, fst_flags):
    io = self.fds.get(fd)
    if io is None or isinstance(io, self.WasiDir):
        return self.ERRNO_BADF
    try:
        st = os.fstat(io.fileno())
    except OSError:
        return self.ERRNO_IO
    times, err = self.fst_times(st, atim, mtim, fst_flags)
    if err is not None:
        return err
    try:
        os.utime(io.fileno(), ns=times)
    except OSError:
        return self.ERRNO_IO
    return self.ERRNO_SUCCESS
